storage/innobase/os/os0file.cc

```diff
@@ -1448,7 +1452,9 @@ os_file_create_simple_no_error_handling_func(
 	if (!srv_read_only_mode
 	    && *success
 	    && (srv_unix_file_flush_method == SRV_UNIX_O_DIRECT
-		|| srv_unix_file_flush_method == SRV_UNIX_O_DIRECT_NO_FSYNC)) {
+		|| srv_unix_file_flush_method == SRV_UNIX_O_DIRECT_NO_FSYNC
+		|| srv_unix_file_flush_method == SRV_UNIX_ALL_O_DIRECT)
```

Comment:
3rd condition (ALL_O_DIRECT) was missing so good we identified it and fixing it now but I see the same condition is handled bit differently in os_file_create_func()

--- Function explicitly separate path for log/data file with needed modes.
--- It may be good idea to just use the same snippet here too (may be convert it to a macro/func)
--- and of-course if os_file_create_simple_no_error_handling is not being called for log file then
--- we may simply add an assert to avoid accidental usage in future.

        if (!srv_read_only_mode
            && *success
            && type != OS_LOG_FILE
            && (srv_unix_file_flush_method == SRV_UNIX_O_DIRECT
                || srv_unix_file_flush_method == SRV_UNIX_O_DIRECT_NO_FSYNC)) {

                os_file_set_nocache(file, name, mode_str);
        } else if (!srv_read_only_mode
            && *success
            && srv_unix_file_flush_method == SRV_UNIX_ALL_O_DIRECT) {
                os_file_set_nocache(file, name, mode_str);
        }